packages/console/src/components/TextInput/index.tsx

```diff
@@ -30,11 +37,38 @@ function TextInput(
     className,
     readOnly,
     type = 'text',
+    isConfidential = false,
     ...rest
   }: Props,
   reference: Ref<Nullable<HTMLInputElement>>
 ) {
   const innerRef = useRef<HTMLInputElement>(null);
+  const [hideContent, setHideContent] = useState(true);
+
+  const toggleHiddenContent = () => {
+    setHideContent((previous) => !previous);
+  };
+
+  const hideContentRef = useMemo(() => {
+    if (isConfidential) {
+      return hideContent && type === 'text';
+    }
+
+    return false;
+  }, [hideContent, isConfidential, type]);
+
+  const suffixRef = useMemo(() => {
+    if (isConfidential) {
+      return (
+        <IconButton onClick={toggleHiddenContent}>
+          {hideContentRef ? <EyeClosed /> : <Eye />}
+        </IconButton>
+      );
+    }
```

Comment:
These `useMemo` usages seem unnecessary to me. Just directly put it in the tsx part